website/guide/symbols.html

```diff
@@ -5,23 +5,27 @@ <h1 id="symbols">Symbols</h1>
 <b>hidden Symbol</b> variant similar to internal strings in Duktape 1.x.
 Hidden Symbols differ from ES2015 Symbols in that they're hidden from
 ordinary Ecmascript code: they can't be created from Ecmascript code,
-won't be enumerated or JSON-serialized, and won't be returned even from
-<code>Object.getOwnPropertyNames()</code>.  Properties with hidden Symbol
+won't be enumerated or JSON-serialized, and won't be returned from
+<code>Object.getOwnPropertyNames()</code> or even
+<code>Object.getOwnPropertySymbols()</code>.  Properties with hidden Symbol
 keys can only be accessed by a direct property read/write when holding a
-reference to a hidden Symbol.</p>
-
-<p>Duktape uses hidden Symbols for various implementation specific purposes,
-such as storing an object's finalizer reference.  User code can also use hidden
-Symbols for its own purposes, e.g. to store hidden state in objects.  User code
-should never try to access Duktape's hidden Symbol keyed properties: the set of
-such properties can change arbitrarily between versions.</p>
+reference to the hidden Symbol.</p>
 
 <p>Symbols of all kinds are represented internally using byte sequences which
 are invalid UTF-8; see
 <a href="https://github.com/svaarala/duktape/blob/master/doc/symbols.rst">symbols.rst</a>
-for the current formats in use.  When C code pushes a string using e.g.
-<code>duk_push_string()</code> and the byte sequence matches an internal
-Symbol format, the string value is automatically interpreted as a Symbol.</p>
+for the current formats in use.  Application hidden Symbols begin with a 0xFF
+byte prefix and are followed by an arbitrary, application selected string.  When
+C code pushes a string using e.g. <code>duk_push_string()</code> and the byte
+sequence matches an internal Symbol format, the string value is automatically
+interpreted as a Symbol.</p>
+
+<p>Duktape also uses hidden Symbols for various implementation specific purposes,
+such as storing an object's finalizer reference.  As of Duktape 2.2 a different
+byte prefix is used for Duktape's hidden Symbols, so the 0xFF prefix is now
+reserved entirely for application use.  Application code should never try to
+access Duktape's hidden Symbol keyed properties: the set of such properties can
+change arbitrarily between versions.</p>
```

Comment:
I feel like the mention of the internal representation would easily be covered via the `doc/symbols.rst` link. Probably only mentioning a user code compatible, internal only change and linking to the release notes would be good enough? Who is the audience for the **Guide**? I would imagine only client API users?

I will work on a simplified Symbols section tonight to hopefully better illustrate my comment.

Reply:
The audience of the website is all Duktape developers, at any stage of their involvement: starting from glancing at the index page, getting started, and later on using as a reference. The line between users and Duktape developers is sometimes fuzzy, because a lot of users tweak the internals, maintain patched forks, etc.

As a general principle:
- The index page tries to provide information necessary to figure out if the library is right for you and to make it easy to try.
- The Guide is an umbrella document which should cover all relevant topics (except engine neutral, general Ecmascript topics) minimally and if necessary to link to Github developer docs (doc/xxx.rst) or Wiki for more discussion.
- The API is a reference document which should be convenient when working on stuff and quickly looking up functions.

These three documents are IMO the most important and the goal should be for them to be clean, easy to read, as short as possible but still covering the necessary topics, polished language, etc. The Wiki and the internal developer documentation have a much lower bar.

I don't the documentation achieves the goal I outlined above at present: it still pretty much has the same structure as when I initially wrote it before Duktape 1.0 release. In particular transitioning from "just read the index page", to Getting Started, to actually writing programs (Programming Model) would benefit from a rewrite.

Reply:
Ok, the goals seem different than what I thought. This looks great then.

Reply:
@svaarala For what it's worth, I really like that you wrote the documentation with potential Duktape contributors in mind.  Makes it much less scary when it becomes necessary to hack away at stuff under the hood :)